TVB support plus improved tutorial examples

hugh-osborne opened the pull request:
Added a new app directory TvbModels which contains source code to generate python shared libraries to be used with The Virtual Brain. Currently, only a very basic LIF model and a Wilson Cowan model which matches that in TVB.

CMakeLists.txt needed to be changed to add the -fPIC (Position Independent Compilation) flag to support the shared library make files.

MPINode and MPINetwork have been augmented to provide a single step evolve function and to add external activity connections to and from nodes (these are  used to pass activity to and from the virtual brain).

The existing WilsonCowan algorithm has been updated to add the refractory term to match TVB.

The examples directory has been updated to fix bugs in the older code and add new mesh style simulations.

--mpi, --openmp and --no_root have been added as available flags to miind.py. The cmake_template has been updated to have MPI and OPENMP set up if required plus included -fext-numeric-literals in the compile flags. This should be checked on other systems if possible but testing on my laptop and work machine, submit.py is now successful "out of the box" even if MPI and OPENMP are enabled.

A small change to demo2D.py to allow the user to define the specific mesh they wish to view

Commit:
Bug fixes to 1D simulation examples and added 2D examples for new tutorial

Commit:
Update compile flags for openmp and numeric literals

Do not commit/push this to dekamps/miind - only for convenience on
hugh-osborne/miind

Commit:
Added additional helper functions to demo2D.py and mesh.py

hugh-osborne committed:
Initial work done in Marseilles

Added -fPIC to base CMakeLists required for SHARED wrapper library
Added PythonWrapper project to libs CMakeLists
Added evolveSingleStep functions and removed static external definition
of _localNodes in MPINetwork and MPINetworkCode
Added access functions to MPINode and MPINodeCode

hugh-osborne committed:
oops!

Commit:
Switched to single network instance with multiple nodes defined by tvb

hugh-osborne committed:
MPI TVB solution - NOT COMPLETE!

Commit:
Wilson Cowan setup.

hugh-osborne committed:
Additional Wilson Cowan functions and setup of tvb PythonWrapper

Commit:
Wilson Cowan in TVB!

hugh-osborne committed:
tvb generic and wilson cowan model cleanup

Commit:
Clean code

Commit:
fixed MPI

Commit:
broke it again

hugh-osborne committed:
Update python cmake_template to support MPI and OpenMP by default. Also added -fext-numeric-literals

hugh-osborne committed:
Updated Examples

hugh-osborne committed:
TVB model udpates

hugh-osborne committed:
update TVB

hugh-osborne committed:
Code cleanup ready for merge to master.

hugh-osborne committed:
Merge pull request #1 from hugh-osborne/tvb

Tvb

hugh-osborne committed:
Reinstated response_analyse.py

hugh-osborne committed:
Update to CMakeLists.txt to allow position independent compilation to facilitate shared libraries

hugh-osborne committed:
User can now enable MPI, OPENMP and disable ROOT as parameters to miind.py

dekamps commented:
Hi Hugh,


Looks cool. Do you need anything from me atm?


Cheers,

Marc

________________________________
From: Hugh Osborne <notifications@github.com>
Sent: 27 February 2018 09:29:22
To: dekamps/miind
Cc: Subscribed
Subject: [dekamps/miind] TVB support plus improved tutorial examples (#29)


Added a new app directory TvbModels which contains source code to generate python shared libraries to be used with The Virtual Brain. Currently, only a very basic LIF model and a Wilson Cowan model which matches that in TVB.

CMakeLists.txt needed to be changed to add the -fPIC (Position Independent Compilation) flag to support the shared library make files.

MPINode and MPINetwork have been augmented to provide a single step evolve function and to add external activity connections to and from nodes (these are used to pass activity to and from the virtual brain).

The existing WilsonCowan algorithm has been updated to add the refractory term to match TVB.

The examples directory has been updated to fix bugs in the older code and add new mesh style simulations.

--mpi, --openmp and --no_root have been added as available flags to miind.py. The cmake_template has been updated to have MPI and OPENMP set up if required plus included -fext-numeric-literals in the compile flags. This should be checked on other systems if possible but testing on my laptop and work machine, submit.py is now successful "out of the box" even if MPI and OPENMP are enabled.

A small change to demo2D.py to allow the user to define the specific mesh they wish to view

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/dekamps/miind/pull/29

Commit Summary

  *   Bug fixes to 1D simulation examples and added 2D examples for new tutorial
  *   Update compile flags for openmp and numeric literals
  *   Added additional helper functions to demo2D.py and mesh.py
  *   Initial work done in Marseilles
  *   oops!
  *   Switched to single network instance with multiple nodes defined by tvb
  *   MPI TVB solution - NOT COMPLETE!
  *   Wilson Cowan setup.
  *   Additional Wilson Cowan functions and setup of tvb PythonWrapper
  *   Wilson Cowan in TVB!
  *   tvb generic and wilson cowan model cleanup
  *   Clean code
  *   fixed MPI
  *   broke it again
  *   Update python cmake_template to support MPI and OpenMP by default. Also added -fext-numeric-literals
  *   Updated Examples
  *   TVB model udpates
  *   update TVB
  *   Code cleanup ready for merge to master.
  *   Merge pull request #1 from hugh-osborne/tvb
  *   Reinstated response_analyse.py
  *   Update to CMakeLists.txt to allow position independent compilation to facilitate shared libraries
  *   User can now enable MPI, OPENMP and disable ROOT as parameters to miind.py

File Changes

  *   M CMakeLists.txt<https://github.com/dekamps/miind/pull/29/files#diff-0> (4)
  *   M apps/CMakeLists.txt<https://github.com/dekamps/miind/pull/29/files#diff-1> (1)
  *   A apps/TvbModels/CMakeLists.txt<https://github.com/dekamps/miind/pull/29/files#diff-2> (2)
  *   A apps/TvbModels/lif/CMakeLists.txt<https://github.com/dekamps/miind/pull/29/files#diff-3> (55)
  *   A apps/TvbModels/lif/tvb_lif.cpp<https://github.com/dekamps/miind/pull/29/files#diff-4> (74)
  *   A apps/TvbModels/wilsoncowan/CMakeLists.txt<https://github.com/dekamps/miind/pull/29/files#diff-5> (59)
  *   A apps/TvbModels/wilsoncowan/tvb_wilsoncowan.cpp<https://github.com/dekamps/miind/pull/29/files#diff-6> (158)
  *   A examples/aexp/aexp.model<https://github.com/dekamps/miind/pull/29/files#diff-7> (2400)
  *   A examples/aexp/aexp.xml<https://github.com/dekamps/miind/pull/29/files#diff-8> (53)
  *   A examples/aexp/aexp_-1_0_0.mat<https://github.com/dekamps/miind/pull/29/files#diff-9> (258784)
  *   A examples/aexp/aexp_1_0_0.mat<https://github.com/dekamps/miind/pull/29/files#diff-10> (258784)
  *   A examples/aexp/aexpanalysis.py<https://github.com/dekamps/miind/pull/29/files#diff-11> (106)
  *   A examples/aexp/aexpcd886d1a-d60a-48f1-807b-b3db8de51e3c.fid~<https://github.com/dekamps/miind/pull/29/files#diff-12> (21)
  *   A examples/aexp/aexpnoa642fee6a-9257-4624-9178-9a404b8bad50.model<https://github.com/dekamps/miind/pull/29/files#diff-13> (27)
  *   A examples/aexp/aexpnoa642fee6a-9257-4624-9178-9a404b8bad50_-1_0_0_0_.mat<https://github.com/dekamps/miind/pull/29/files#diff-14> (1019)
  *   A examples/aexp/aexpnoa642fee6a-9257-4624-9178-9a404b8bad50_1_0_0_0_.mat<https://github.com/dekamps/miind/pull/29/files#diff-15> (1019)
  *   A examples/aexp/izsh.model<https://github.com/dekamps/miind/pull/29/files#diff-16> (439)
  *   A examples/aexp/izsh_0.03_0_0.mat<https://github.com/dekamps/miind/pull/29/files#diff-17> (3417)
  *   A examples/cond/cond.model<https://github.com/dekamps/miind/pull/29/files#diff-18> (1388)
  *   A examples/cond/cond.xml<https://github.com/dekamps/miind/pull/29/files#diff-19> (46)
  *   A examples/cond/cond_0_0.005_0.mat<https://github.com/dekamps/miind/pull/29/files#diff-20> (96860)
  *   A examples/cond/cond_0_0.05_0.mat<https://github.com/dekamps/miind/pull/29/files#diff-21> (96860)
  *   A examples/cond/condanalysis.py<https://github.com/dekamps/miind/pull/29/files#diff-22> (138)
  *   A examples/cond/mc_gain_curves.py<https://github.com/dekamps/miind/pull/29/files#diff-23> (57)
  *   A examples/lif/lif.mesh.bak<https://github.com/dekamps/miind/pull/29/files#diff-24> (7)
  *   A examples/lif/lif.model<https://github.com/dekamps/miind/pull/29/files#diff-25> (23)
  *   A examples/lif/lif.xml<https://github.com/dekamps/miind/pull/29/files#diff-26> (46)
  *   A examples/lif/lif_0.005_0_0.mat<https://github.com/dekamps/miind/pull/29/files#diff-27> (601)
  *   A examples/lif/lifanalysis.py<https://github.com/dekamps/miind/pull/29/files#diff-28> (212)
  *   A examples/lif/steady_state.dat<https://github.com/dekamps/miind/pull/29/files#diff-29> (10000)
  *   A examples/qif/qif.model<https://github.com/dekamps/miind/pull/29/files#diff-30> (29)
  *   A examples/qif/qif.xml<https://github.com/dekamps/miind/pull/29/files#diff-31> (59)
  *   A examples/qif/qif_-0.05_0_0.mat<https://github.com/dekamps/miind/pull/29/files#diff-32> (1722)
  *   A examples/qif/qif_0.005_0_0.mat<https://github.com/dekamps/miind/pull/29/files#diff-33> (1722)
  *   A examples/qif/qif_0.03_0_0.mat<https://github.com/dekamps/miind/pull/29/files#diff-34> (1722)
  *   A examples/qif/qif_0.05_0_0.mat<https://github.com/dekamps/miind/pull/29/files#diff-35> (1722)
  *   A examples/qif/qif_0.15_0_0.mat<https://github.com/dekamps/miind/pull/29/files#diff-36> (1722)
  *   A examples/qif/qif_excit.model<https://github.com/dekamps/miind/pull/29/files#diff-37> (29)
  *   A examples/qif/qif_excit_-0.05_0_0.mat<https://github.com/dekamps/miind/pull/29/files#diff-38> (1183)
  *   A examples/qif/qif_excit_0.05_0_0.mat<https://github.com/dekamps/miind/pull/29/files#diff-39> (1183)
  *   A examples/qif/qif_inhib.model<https://github.com/dekamps/miind/pull/29/files#diff-40> (29)
  *   A examples/qif/qif_inhib_-0.05_0_0.mat<https://github.com/dekamps/miind/pull/29/files#diff-41> (358)
  *   A examples/qif/qif_inhib_0.05_0_0.mat<https://github.com/dekamps/miind/pull/29/files#diff-42> (358)
  *   A examples/qif/qifanalysis.py<https://github.com/dekamps/miind/pull/29/files#diff-43> (171)
  *   A examples/qif/show_qif_gain.py<https://github.com/dekamps/miind/pull/29/files#diff-44> (131)
  *   M examples/rate_based/rate.xml<https://github.com/dekamps/miind/pull/29/files#diff-45> (34)
  *   D examples/response_curve_lif/analytic.dat<https://github.com/dekamps/miind/pull/29/files#diff-46> (2401)
  *   M examples/response_curve_lif/response.xml<https://github.com/dekamps/miind/pull/29/files#diff-47> (8)
  *   M examples/single_lif/analyse_single_lif.py<https://github.com/dekamps/miind/pull/29/files#diff-48> (18)
  *   A examples/single_lif/no_decay.xml<https://github.com/dekamps/miind/pull/29/files#diff-49> (59)
  *   A examples/single_lif/no_input.xml<https://github.com/dekamps/miind/pull/29/files#diff-50> (59)
  *   M examples/single_lif/response.xml<https://github.com/dekamps/miind/pull/29/files#diff-51> (12)
  *   M examples/single_lif/single_lif.py<https://github.com/dekamps/miind/pull/29/files#diff-52> (6)
  *   M examples/single_selfinhib/self.xml<https://github.com/dekamps/miind/pull/29/files#diff-53> (32)
  *   M examples/twopop/twopop.xml<https://github.com/dekamps/miind/pull/29/files#diff-54> (18)
  *   M libs/CMakeLists.txt<https://github.com/dekamps/miind/pull/29/files#diff-55> (4)
  *   M libs/MPILib/include/MPINetwork.hpp<https://github.com/dekamps/miind/pull/29/files#diff-56> (49)
  *   M libs/MPILib/include/MPINetworkCode.hpp<https://github.com/dekamps/miind/pull/29/files#diff-57> (168)
  *   M libs/MPILib/include/MPINode.hpp<https://github.com/dekamps/miind/pull/29/files#diff-58> (36)
  *   M libs/MPILib/include/MPINodeCode.hpp<https://github.com/dekamps/miind/pull/29/files#diff-59> (56)
  *   A libs/MPILib/include/MiindTvbModelAbstract.hpp<https://github.com/dekamps/miind/pull/29/files#diff-60> (96)
  *   M libs/MPILib/include/WilsonCowanParameter.hpp<https://github.com/dekamps/miind/pull/29/files#diff-61> (9)
  *   M libs/MPILib/src/WilsonCowanAlgorithm.cpp<https://github.com/dekamps/miind/pull/29/files#diff-62> (39)
  *   M python/cmake_template<https://github.com/dekamps/miind/pull/29/files#diff-63> (33)
  *   M python/demo2D.py<https://github.com/dekamps/miind/pull/29/files#diff-64> (9)
  *   M python/directories.py<https://github.com/dekamps/miind/pull/29/files#diff-65> (40)
  *   M python/mesh.py<https://github.com/dekamps/miind/pull/29/files#diff-66> (19)
  *   M python/miind.py<https://github.com/dekamps/miind/pull/29/files#diff-67> (11)
  *   A python/miindTVBlibs.py<https://github.com/dekamps/miind/pull/29/files#diff-68> (7)

Patch Links:

  *   https://github.com/dekamps/miind/pull/29.patch
  *   https://github.com/dekamps/miind/pull/29.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/dekamps/miind/pull/29>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AEk-YzFrbpOT3lWvSS3eG9qED3q7t26wks5tY8rygaJpZM4SUm1Q>.

hugh-osborne committed:
Introduce MiindIO

Commit:
Slightly more nuanced functionality for Variable in xml plus cleaner miind_api

hugh-osborne committed:
continuing work on miind api and miind ui

hugh-osborne committed:
More API work

hugh-osborne committed:
rinzel mesh

Commit:
more rinzel

hugh-osborne committed:
more

Commit:
proper dodgy attempt to destroy Monte Carlo

hugh-osborne committed:
ooop?

hugh-osborne committed:
ooop2?

Commit:
more ui clean up

Commit:
good api stuff

hugh-osborne committed:
fixed generator message

hugh-osborne committed:
api clean up

Commit:
changes

Commit:
Merge branch 'miindapi' of https://github.com/hugh-osborne/miind into miindapi

Commit:
blah

hugh-osborne committed:
fix Reversal

Commit:
reinstate calculated overlaps

Commit:
support for "submitting" xml for shared library

Commit:
better fid tools (from mikkel)

hugh-osborne committed:
temp fix for matrix gen geometric solution

Commit:
fixes for shared library generation

Commit:
External Connections for Shared Library

Commit:
fixes

Commit:
api fixes

Commit:
miind_ui renamed to miindio. Can now be called once with a command as a parameter. NICE!

Commit:
fix

Commit:
tidy up

Commit:
python example

Commit:
Bug fixes

Commit:
bug_fixes

Commit:
cmake template for python lib

hugh-osborne committed:
ready for merge

hugh-osborne committed:
Matrix Generator has option to use quad area calculation instead of monte carlo.

MeshAlgorithmCode writes rates as it goes to files that aren't root.

Miindio rates now come from the written rate files.

Transition generator area calculation uses monte carlo for cells with lost mass > 5% so we can jump across gaps.

hugh-osborne committed:
fixes

Commit:
Fixes.

Commit:
Added docker stuff and removed monte carlo backup from quad overlap calculation in MatGen

hugh-osborne committed:
generate.

hugh-osborne committed:
fixed mass loss?

hugh-osborne committed:
faster.

hugh-osborne committed:
oops

dekamps committed:
Changes to be committed:
	modified:   apps/CMakeLists.txt
	modified:   apps/UnitTwoDLib/CMakeLists.txt
	modified:   apps/UnitTwoDLib/MeshAlgorithmTest.cpp
	modified:   apps/UnitTwoDLib/MeshTest.cpp
	modified:   apps/UnitTwoDLib/QuadrilateralTest.cpp
	modified:   libs/TwoDLib/CMakeLists.txt
	modified:   libs/TwoDLib/CSRMatrix.cpp
	modified:   libs/TwoDLib/CSRMatrix.hpp
	modified:   libs/TwoDLib/MasterOMP.cpp
	modified:   libs/TwoDLib/MasterOdeint.cpp
	modified:   libs/TwoDLib/MasterOdeint.hpp
	modified:   libs/TwoDLib/MeshAlgorithm.hpp
	modified:   libs/TwoDLib/MeshAlgorithmCode.hpp
	modified:   libs/TwoDLib/Ode2DSystem.cpp
	modified:   libs/TwoDLib/Ode2DSystem.hpp
	modified:   libs/TwoDLib/TransitionMatrix.cpp
	modified:   libs/TwoDLib/TransitionMatrix.hpp
	modified:   libs/TwoDLib/TwoDLib.hpp

dekamps committed:
new file:   ../../../apps/IntegrationTwoDLib/aexp97b7e1e7-b88c-4ff8-8e86-7e1e092f4d9a.model
	new file:   ../../../apps/IntegrationTwoDLib/aexp97b7e1e7-b88c-4ff8-8e86-7e1e092f4d9a_-1_0_0_0_.mat
	new file:   ../../../apps/IntegrationTwoDLib/aexp97b7e1e7-b88c-4ff8-8e86-7e1e092f4d9a_1_0_0_0_.mat
	new file:   ../../../apps/IntegrationTwoDLib/aexp_brette.cpp
	new file:   ../../../apps/IntegrationTwoDLib/condee2a5ff4-0087-4d69-bae3-c0a223d03693.mat
	new file:   ../../../apps/IntegrationTwoDLib/condee2a5ff4-0087-4d69-bae3-c0a223d03693.model
	new file:   ../../../apps/IntegrationTwoDLib/condee2a5ff4-0087-4d69-bae3-c0a223d03693_0_0.05_0_0_.mat
	modified:   ../../../apps/UnitTwoDLib/CMakeLists.txt
	modified:   ../../../libs/MPILib/src/report/handler/MinimalReportHandler.cpp
	modified:   ../../../libs/TwoDLib/CSRMatrix.cpp
	modified:   ../../../libs/TwoDLib/CSRMatrix.hpp
	modified:   ../../../libs/TwoDLib/MasterOMP.cpp
	modified:   ../../../libs/TwoDLib/MasterOMP.hpp
	modified:   ../../../libs/TwoDLib/MasterOdeint.cpp
	modified:   ../../../libs/TwoDLib/MasterOdeint.hpp
	modified:   ../../../libs/TwoDLib/Mesh.cpp
	modified:   ../../../libs/TwoDLib/MeshAlgorithm.hpp
	modified:   ../../../libs/TwoDLib/MeshAlgorithmCode.hpp

dekamps committed:
new file:   CMakeLists.txt

dekamps committed:
new file:   condee2a5ff4-0087-4d69-bae3-c0a223d03693.stat

dekamps committed:
new file:   condee2a5ff4-0087-4d69-bae3-c0a223d03693.res
	new file:   condee2a5ff4-0087-4d69-bae3-c0a223d03693.rev

dekamps committed:
new file:   cond.cpp
	new file:   ../UnitTwoDLib/CSRTest.cpp
	new file:   ../UnitTwoDLib/Ode2DSystemGroupTest.cpp

dekamps committed:
new file:   ../../libs/TwoDLib/Ode2DSystemGroup.cpp
	new file:   ../../libs/TwoDLib/Ode2DSystemGroup.hpp

dekamps committed:
new file:   MasterGroupTest.cpp
	new file:   MeshAlgorithmGroupTest.cpp

dekamps committed:
modified:   ../../libs/TwoDLib/MeshAlgorithm.hpp

dekamps committed:
modified:   apps/IntegrationTwoDLib/CMakeLists.txt
	modified:   libs/TwoDLib/MeshAlgorithmCode.hpp

dekamps committed:
modified:   ../../../CMakeLists.txt
	modified:   ../../../apps/IntegrationTwoDLib/MeshAlgorithmGroupTest.cpp
	modified:   ../../../apps/MatrixGenerator/Bind.cpp
	modified:   ../../../apps/MatrixGenerator/GenerateMatrix.cpp
	modified:   ../../../apps/MatrixGenerator/TranslationObject.cpp
	modified:   ../../../apps/Projection/Projection.cpp
	modified:   ../../../apps/UnitTwoDLib/CMakeLists.txt
	modified:   ../../../apps/UnitTwoDLib/CSRTest.cpp
	modified:   ../../../apps/UnitTwoDLib/MeshTest.cpp
	modified:   ../../../apps/UnitTwoDLib/Ode2DSystemGroupTest.cpp
	modified:   ../../../apps/UnitTwoDLib/Ode2DSystemTest.cpp
	modified:   ../../../apps/UnitTwoDLib/TransitionMatrixTest.cpp
	modified:   ../../../libs/TwoDLib/CSRMatrix.cpp
	modified:   ../../../libs/TwoDLib/CheckSystem.cpp
	modified:   ../../../libs/TwoDLib/Mesh.hpp
	modified:   ../../../libs/TwoDLib/Ode2DSystem.cpp
	modified:   ../../../libs/TwoDLib/Ode2DSystemGroup.cpp
	modified:   ../../../libs/TwoDLib/Ode2DSystemGroup.hpp
	modified:   ../../../libs/TwoDLib/TransitionMatrix.cpp
	modified:   ../../../libs/TwoDLib/TransitionMatrixGenerator.cpp

dekamps committed:
modified:   ../../../apps/IntegrationTwoDLib/MeshAlgorithmGroupTest.cpp
	modified:   ../../../libs/TwoDLib/Ode2DSystemGroup.hpp

dekamps committed:
new file:   FixtureOde2DSystemGroup.cpp
	new file:   FixtureOde2DSystemGroup.hpp

dekamps committed:
modified:   CMakeLists.txt

dekamps committed:
modified:   ../../../apps/IntegrationTwoDLib/MeshAlgorithmGroupTest.cpp
	modified:   ../../../apps/UnitTwoDLib/FixtureOde2DSystemGroup.cpp
	modified:   ../../../apps/UnitTwoDLib/FixtureOde2DSystemGroup.hpp
	modified:   ../../../apps/UnitTwoDLib/Ode2DSystemGroupTest.cpp
	modified:   ../../../libs/TwoDLib/CSRMatrix.hpp
	modified:   ../../../libs/TwoDLib/Ode2DSystemGroup.cpp

dekamps committed:
modified:   CMakeLists.txt
	new file:   Euler.cpp
	new file:   Euler.cpp~
	new file:   Euler.hpp
	new file:   Euler.hpp~
	modified:   MeshAlgorithmGroupTest.cpp
	new file:   rinzel.cpp
	new file:   rinzel.cpp~
	new file:   rinzel.model
	modified:   ../../libs/TwoDLib/CSRMatrix.hpp
	modified:   ../../libs/TwoDLib/MeshAlgorithm.hpp

dekamps committed:
new file:   rinzel_-0.1_0_0_0_.mat
	new file:   rinzel_0.1_0_0_0_.mat

hugh-osborne committed:
Support (initial) for refractory period plus working rybak model?!

hugh-osborne committed:
better refract

dekamps committed:
modified:   CMakeLists.txt
	modified:   apps/IntegrationTwoDLib/CMakeLists.txt
	modified:   apps/IntegrationTwoDLib/MasterGroupTest.cpp
	modified:   apps/IntegrationTwoDLib/MeshAlgorithmGroupTest.cpp
	modified:   apps/IntegrationTwoDLib/rinzel.cpp
	modified:   apps/IntegrationTwoDLib/rinzel.cpp~
	modified:   libs/MPILib/include/TypeDefinitions.hpp
	modified:   libs/NumtoolsLib/Distribution.cpp
	modified:   libs/NumtoolsLib/RandomGenerator.cpp
	modified:   libs/NumtoolsLib/RandomGenerator.h
	modified:   libs/TwoDLib/CMakeLists.txt
	modified:   libs/TwoDLib/CSRMatrix.hpp
	renamed:    apps/IntegrationTwoDLib/Euler.cpp -> libs/TwoDLib/Euler.cpp
	renamed:    apps/IntegrationTwoDLib/Euler.cpp~ -> libs/TwoDLib/Euler.cpp~
	renamed:    apps/IntegrationTwoDLib/Euler.hpp~ -> libs/TwoDLib/Euler.hpp
	renamed:    apps/IntegrationTwoDLib/Euler.hpp -> libs/TwoDLib/Euler.hpp~
	modified:   libs/TwoDLib/MasterOMP.cpp
	modified:   libs/TwoDLib/MeshAlgorithmCode.hpp
	modified:   libs/TwoDLib/Ode2DSystemGroup.cpp
	modified:   libs/TwoDLib/Ode2DSystemGroup.hpp
	modified:   libs/TwoDLib/Quadrilateral.cpp
	modified:   libs/TwoDLib/TwoDLib.hpp
	modified:   python/algorithms.py
	modified:   python/cmake_template
	modified:   python/miind.py
	modified:   python/parametersweep.py

dekamps committed:
new file:   directories3.py
	modified:   miind.py

dekamps committed:
new file:   codegen3.py

dekamps committed:
modified:   submit.py

dekamps committed:
modified:   CMakeLists.txt
	modified:   MeshAlgorithm.hpp
	modified:   MeshAlgorithmCode.hpp
	modified:   TwoDLib.hpp
	new file:   XML.hpp
	modified:   ../../python/miind.py

dekamps committed:
modified:   CMakeLists.txt

dekamps committed:
modified:   libs/TwoDLib/Ode2DSystemGroup.cpp
	modified:   libs/TwoDLib/Ode2DSystemGroup.hpp

dekamps committed:
modified:   libs/TwoDLib/Ode2DSystemGroup.cpp

dekamps committed:
new file:   CMakeLists.txt
	new file:   CSRAdapter.cu
	new file:   CSRAdapter.cuh
	new file:   CudaEuler.cu
	new file:   CudaEuler.cuh
	new file:   CudaOde2DSystemAdapter.cu
	new file:   CudaOde2DSystemAdapter.cuh
	new file:   CudaTwoDLib.hpp

dekamps committed:
modified:   ../CMakeLists.txt
	new file:   AdapterTest.cpp
	new file:   CudaUnit.cpp
	new file:   FixtureAdapter.cpp
	new file:   FixtureAdapter.hpp
	new file:   MatrixTest.cpp
	modified:   ../../libs/CMakeLists.txt

Commit:
part implemented refractory period plus python library fixes and improvements

dekamps committed:
modified:   apps/UnitCudaLib/AdapterTest.cpp
	modified:   apps/UnitTwoDLib/CMakeLists.txt
	modified:   libs/CudaTwoDLib/CudaOde2DSystemAdapter.cu
	modified:   libs/CudaTwoDLib/CudaOde2DSystemAdapter.cuh
	modified:   libs/TwoDLib/CMakeLists.txt
	modified:   libs/TwoDLib/Euler.hpp
	modified:   libs/TwoDLib/TwoDLib.hpp
	modified:   python/miind.py

dekamps committed:
new file:   bary3.py
	new file:   mesh3.py
	new file:   meshtest3.py

dekamps committed:
new file:   condee2a5ff4-0087-4d69-bae3-c0a223d03693.model
	new file:   condee2a5ff4-0087-4d69-bae3-c0a223d03693_0_0.05_0_0_.mat

dekamps committed:
new file:   CSRAdapter.cpp
	new file:   CSRAdapter.hpp
	new file:   CSRAdapter.hpp~

dekamps committed:
new file:   AdapterTest.cpp

dekamps committed:
modified:   CudaEmulationTest.cpp

dekamps committed:
modified:   CudaEmulationTest.cpp

dekamps committed:
modified:   CudaEmulationTest.cpp

dekamps committed:
modified:   CudaEmulationTest.cpp

dekamps committed:
modified:   CudaEmulationTest.cpp

dekamps committed:
modified:   CudaEmulationTest.cpp

dekamps committed:
modified:   CudaEmulationTest.cpp

dekamps committed:
modified:   CudaEmulationTest.cpp

dekamps committed:
modified:   ../libs/CudaTwoDLib/CudaOde2DSystemAdapter.cu
	modified:   ../libs/CudaTwoDLib/CudaOde2DSystemAdapter.cuh
	modified:   ../python/cmake_template
	modified:   ../python/miind.py

dekamps committed:
modified:   python/miind.py

dekamps committed:
modified:   python/miind.py

dekamps committed:
modified:   python/miind.py

dekamps committed:
modified:   python/miind.py

dekamps committed:
modified:   python/miind.py

dekamps committed:
modified:   python/miind.py

dekamps committed:
modified:   ../apps/UnitTwoDLib/AdapterTest.cpp
	modified:   ../libs/CudaTwoDLib/CudaOde2DSystemAdapter.cu
	modified:   ../libs/CudaTwoDLib/CudaOde2DSystemAdapter.cuh

dekamps committed:
modified:   ../libs/CudaTwoDLib/CudaOde2DSystemAdapter.cu
	modified:   ../libs/CudaTwoDLib/CudaOde2DSystemAdapter.cuh

dekamps committed:
modified:   ../libs/CudaTwoDLib/CudaOde2DSystemAdapter.cuh

dekamps committed:
modified:   ../python/miind.py

dekamps committed:
modified:   ../python/miind.py

dekamps committed:
modified:   CMakeLists.txt

dekamps committed:
modified:   CMakeLists.txt

dekamps committed:
modified:   AdapterTest.cpp
modified:   ../../libs/CudaTwoDLib/CudaOde2DSystemAdapter.cuh

dekamps committed:
modified:   AdapterTest.cpp

dekamps committed:
modified:   CMakeLists.txt
	modified:   ../../python/cmake_template

dekamps committed:
modified:   CMakeLists.txt

dekamps committed:
modified:   cmake_template
	modified:   directories3.py

dekamps committed:
modified:   directories3.py

dekamps committed:
modified:   python/cmake_template
	modified:   python/directories3.py
	modified:   python/miind.py

dekamps committed:
modified:   python/directories3.py

dekamps committed:
modified:   python/cmake_template
	modified:   python/directories3.py
	modified:   python/miind.py

dekamps committed:
Changes to be committed:
	modified:   libs/CudaTwoDLib/CudaOde2DSystemAdapter.cu
	modified:   libs/CudaTwoDLib/CudaOde2DSystemAdapter.cuh
	modified:   libs/TwoDLib/Ode2DSystemGroup.cpp
	modified:   python/miind.py

dekamps committed:
modified:   miind.py

dekamps committed:
modified:   python/miind.py

dekamps committed:
modified:   CMakeLists.txt

dekamps committed:
modified:   ../libs/CudaTwoDLib/CSRAdapter.cu

hugh-osborne committed:
remove WithK2 example, small density fix

hugh-osborne committed:
merge fix

dekamps committed:
modified:   ../libs/CudaTwoDLib/CSRAdapter.cu
modified:   ../libs/CudaTwoDLib/CSRAdapter.cuh
modified:   ../libs/TwoDLib/CSRMatrix.hpp

dekamps committed:
modified:   ../libs/CudaTwoDLib/CSRAdapter.cu
modified:   ../libs/TwoDLib/CSRMatrix.hpp

dekamps committed:
modified:   ../libs/CudaTwoDLib/CSRAdapter.cu

dekamps committed:
modified:   ../libs/CudaTwoDLib/CSRAdapter.cu

hugh-osborne committed:
merge from dekamps master

dekamps committed:
modified:   libs/CudaTwoDLib/CudaEuler.cu

dekamps committed:
modified:   libs/CudaTwoDLib/CudaEuler.cu

Commit:
Added support for jump files in Matrix Gen, plus command to only change the reset mapping.

Commit:
Error Fix in MeshTools

hugh-osborne committed:
Fix probability reporting with tau_refractive

Commit:
bug fixes

Commit:
Grid IN A DAY WHAAAAAAT

hugh-osborne committed:
reset mapping stuff for transform matrix

hugh-osborne committed:
rate plot fix

Commit:
sneaky alternative Master Equation solver

Commit:
merge oops

Commit:
hmm, not working...

hugh-osborne committed:
YES - working and super fast

hugh-osborne committed:
OPENGL

hugh-osborne committed:
oops needed the new files OPENGL

hugh-osborne committed:
grrr

hugh-osborne committed:
axes in opengl

hugh-osborne committed:
separate Grid Algorithm into proper class

hugh-osborne committed:
clean up

hugh-osborne committed:
getting there

hugh-osborne committed:
auto grid gen

hugh-osborne committed:
opengl fix and grid generation

Commit:
fix

hugh-osborne committed:
blah

hugh-osborne committed:
better

hugh-osborne committed:
crash fix

Commit:
mutex on display

Commit:
Lovely display stuff...

hugh-osborne committed:
reporting singleton

Commit:
display and report works with MPI

Commit:
cond ffs

hugh-osborne committed:
different v

hugh-osborne committed:
Proper master equation + grid generation fix

hugh-osborne committed:
fixed

hugh-osborne committed:
firing rate fix?

Commit:
hmmmmm

hugh-osborne committed:
something about buffer

Commit:
merge from upstream develop

Commit:
merge from grid

Commit:
got ya

Commit:
wtf

hugh-osborne committed:
fixed overlap

Commit:
blargh so many bugs

Commit:
better still not quite

hugh-osborne committed:
changes

Commit:
blah

hugh-osborne committed:
ok fixed for both v and w directions

hugh-osborne committed:
Lukas' draw-mesh fix

Commit:
cuda 2

Commit:
Fitzhugh Nagumo

Commit:
compile fixes

Commit:
Lukas' bug fixes

Commit:
Reporting part 1

hugh-osborne committed:
merge maybe?

hugh-osborne committed:
hmmm waiting on Marc for final code

hugh-osborne committed:
Easier Reporting

hugh-osborne committed:
Merge branch 'grid'

hugh-osborne committed:
not there yet

hugh-osborne committed:
more

hugh-osborne committed:
losing mass...

hugh-osborne committed:
ooo done

hugh-osborne committed:
Support in VectorizedNetwork for both meshes and grids - hopefully

hugh-osborne committed:
Dammit, Marc, putting ideas in my head.

hugh-osborne committed:
Added parallel reset - required for grid

hugh-osborne committed:
Added back the indeed stuff, maybe it could work

hugh-osborne committed:
derivative calculation without offsets

dekamps committed:
modified:   mesh.py

dekamps committed:
modified:   CMakeLists.txt

dekamps committed:
Merge https://github.com/dekamps/miind into develop

	apps/UnitCudaLib/CMakeLists.txt
	libs/TwoDLib/CSRMatrix.hpp
	libs/TwoDLib/MeshAlgorithm.hpp
	libs/TwoDLib/MeshAlgorithmCode.hpp

dekamps committed:
modified:   ../../libs/TwoDLib/MeshAlgorithm.hpp

hugh-osborne committed:
Implemented refractory period in cuda code

hugh-osborne committed:
merge from develop branch - missing some files

dekamps committed:
new file:   PolyGenerator.cpp
	new file:   PolyGenerator.hpp

dekamps committed:
new file:   timefactor.mesh

hugh-osborne committed:
python setup plus system group support in grid algorithm

hugh-osborne committed:
Merge branch 'develop' of https://github.com/dekamps/miind

hugh-osborne committed:
Refractory period in both PCU and GPU modes plus python generation of vectorized code

hugh-osborne committed:
tested mesh - seems to work for both although there is a max min mass descrepancy

hugh-osborne committed:
Support for both mesh and grid nodes in the same vectorized network

hugh-osborne committed:
Implemented delay for vectorized network and both Grid and MeshAlgorithms - moving forward, the algorithm itself, if it supports DelayedConnection should implement the queue for a delay

hugh-osborne committed:
Python3 support across miindio, vectorized code, normal functionality, python libraries - some bugs to fix in python vectorized.

hugh-osborne committed:
vectorized python fixed. ENABLE_CUDA flag added to miind cmake build. Miindio settings now inferred from CMakeCache if available.

hugh-osborne committed:
Updated GLM for later versions of gcc, VectorizedNetwork supports any order of node in XML code, examples updated.

hugh-osborne committed:
Fixed build error when OPENMP disabled. Fixed error with ROOT disabled for old rate algorithms

hugh-osborne committed:
Fixed missing mass bug in vectorized when not synchronising device at the end of some kernel calls. Fixed rate command in miindio when nodes in unexpected order. Fixed Matrix Generator Bug where the efficacy spans multiple cells and goes beyond threshold (Lukas)

hugh-osborne committed:
Fixed race condition in vectorized network - streams now only shared among meshes, not transition matrices. Additional fix for Lukas which needs testing.

hugh-osborne committed:
Hopefully, all concurrency issues are sorted in vectorized version now. There is still a small difference between CPU vs GPU but I suspect this is because GPU uses float and CPU uses double.

hugh-osborne committed:
Fixed various refractory and delay bugs. Added Jump functionality to grid method. GridAlgorithms now take CustomConnectionParameters as the WeightType to allow for different styles of input connection like with Jump files

hugh-osborne committed:
python updated to generate cpp code which works in docker. Removed package details. Only Dockerfile required.

hugh-osborne committed:
GridAlgorithm Motoneurons and half of vectorized version

hugh-osborne committed:
Refractory period fixed. Rolled back vectorized motoneuron support for now

hugh-osborne committed:
backup

hugh-osborne committed:
Added nice grid examples

hugh-osborne committed:
slight python3 bug in api tools

hugh-osborne committed:
Include Mesh Algoirthm for custom connection paramters - would be nice if it wasn't just a CnP of the original

hugh-osborne committed:
grid tweaks

hugh-osborne committed:
t_interval fix for examples, fix to python code generation to use correct display node list

hugh-osborne committed:
Fixed disagreement between vectorized and normal, added back the support for CustomParameters in vectorised Mesh

hugh-osborne committed:
Bug fixes and updates to examples - lif_python example fixed

hugh-osborne committed:
Removed Boost Python - by disabling MPI, we can eliminate having to build boost altogether.

hugh-osborne committed:
Remove requirement for Boost PYthon. Fixed command history bug in miindio

hugh-osborne committed:
Merge from mn_vectorized : Various bug fixes, removal of boost python, additional gridalgorithm versions and support for CustomParameterConnection in mesh and grid

hugh-osborne committed:
Updated all CMakeLists to generate and install all headers/libs

hugh-osborne committed:
install CMakeLists updated

hugh-osborne committed:
CMakeLists install set up

hugh-osborne committed:
build bugs
